src/parquet/arrow/reader.cc

```diff
@@ -932,6 +1149,22 @@ Status PrimitiveImpl::NextBatch(int64_t records_to_read, std::shared_ptr<Array>*
     TRANSFER_CASE(DATE32, ::arrow::Date32Type, Int32Type)
     TRANSFER_CASE(DATE64, ::arrow::Date64Type, Int32Type)
     TRANSFER_CASE(FIXED_SIZE_BINARY, ::arrow::FixedSizeBinaryType, FLBAType)
+    case ::arrow::Type::DECIMAL: {
+      switch (descr_->physical_type()) {
+        case ::parquet::Type::INT32: {
+          TRANSFER_DATA(::arrow::DecimalType, Int32Type);
+        } break;
+        case ::parquet::Type::INT64: {
+          TRANSFER_DATA(::arrow::DecimalType, Int64Type);
+        } break;
+        case ::parquet::Type::FIXED_LEN_BYTE_ARRAY: {
+          TRANSFER_DATA(::arrow::DecimalType, FLBAType);
+        } break;
```

Comment:
Do any systems that we know of use BYTE_ARRAY?

Reply:
I'm not aware of any. parquet-mr *does* support this, but both hive and spark always use `FIXED_LEN_BYTE_ARRAY`. (spark uses int32 or int64, optionally, in later versions). I can implement this now, or in a follow up patch.